Documentation updates -- moving content from the wiki (and elsewhere)

ryscher opened the pull request:
For https://github.com/CDL-Dryad/dryad-product-roadmap/issues/642

Moving some technical documentation from the old wiki into this folder, in preparation for decommissioning the wiki server.

sfisher commented on documentation/submission_flow.md:
```diff
@@ -20,11 +20,12 @@ Basic Dryad Submission Flow
+5. Notifying of completion
+    * When Merritt has successfully ingested the dataset, the dataset
+      shows up in Merritt's the OAI-PMH feed.
+    * The stash-notifier runs every 5 minutes and checks for the
```
Super minor comment.  Now runs every 1 minute.  I don't think it's critical to change if it's a hassle.

sfisher reviewed (approved):
Looks good to me.  Thanks for going through all this, @ryscher .

ryscher commented:
Ok, I changed the note about the notifier frequency and added a few more things from the Confluence page. The information about deleting datasets is still in Confluence, since it contains some more specific details that we might not want public.
